pkg/macaroon/apikey.go

```diff
@@ -137,8 +137,8 @@ func (a *APIKey) Tail() []byte {
 }
 
 // Serialize serializes the API Key to a string
-func (a *APIKey) Serialize() (string, error) {
-	return base58.CheckEncode(a.mac.Serialize(), 0), nil
+func (a *APIKey) Serialize() string {
```

Comment:
When I reviewed this code, I thought that returning always a `nil` error was because a future change of the implementation which may return some error was not going to break the exported API.

I would have probably asked rather than assuming it.

Reply:
that was the reason but i don't care strongly

Reply:
We also have this public exported interface in lib/uplink that doesn't error: https://godoc.org/storj.io/storj/lib/uplink#APIKey.Serialize

That informed my decision to remove the error, because if the APIKey type includes a macaroon, it will have to ignore it.

Reply:
i'm just a dummy! it's okay to add it there too. very minor changes are okay for lib/uplink still. but either way

Reply:
Honestly, I'd prefer to not have serialize be able to error. How important is this?

Reply:
Being a method it shouldn't report an error except that an instance of such types could not be serialized for some reason (e.g. zero value isn't serializable).

If we don't envision that a future change could return an error, then it looks good to me to remove it.